modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -2328,19 +2345,25 @@ private CompletableFuture<Boolean> ensureReplicaIsPrimary(ReplicaRequest request
         }
 
         if (expectedTerm != null) {
-            return raftClient.refreshAndGetLeaderWithTerm()
-                    .thenCompose(replicaAndTerm -> {
-                                long currentTerm = replicaAndTerm.term();
-
-                                if (expectedTerm == currentTerm) {
-                                    return completedFuture(null);
+            return placementDriver.getPrimaryReplica(replicationGroupId, hybridClock.now().addPhysicalTime(HybridTimestamp.CLOCK_SKEW))
+                    .thenCompose(primaryReplica -> {
+                                long currentEnlistmentConsistencyToken = primaryReplica.getStartTime().longValue();
+
+                                if (expectedTerm.equals(currentEnlistmentConsistencyToken)) {
+                                    if (primaryReplica.getExpirationTime().before(hybridClock.now())) {
+                                        return failedFuture(
+                                                new PrimaryReplicaMissException(expectedTerm, currentEnlistmentConsistencyToken));
```

Comment:
are you going to add TODOs here to rename `term` variables?

Reply:
Yep, added https://issues.apache.org/jira/browse/IGNITE-20377